### PR TITLE
ENG-1519, ENG-1372, ENG-1511, ENG-1768, ENG-1516: 2.11.0 functions

### PR DESCRIPTION
**Pull request by trevorsibanda**

### CODE REVIEW

File: faunadb/client_test.go [hunk 1]
```diff
@@ -941,6 +941,51 @@ func (s *ClientTestSuite) TestEvalCasefoldExpression() {
...
+
+func (s *ClientTestSuite) TestEvalEndsWithExpression() {
+	var b bool
+
+	s.queryAndDecode(f.EndsWith("faunadb", "fauna"), &b)
+	s.Require().Equal(false, b)
+
+	s.queryAndDecode(f.EndsWith("faunadb", ""), &b)
+	s.Require().Equal(true, b)
+}
```
Comment (evbruno) [inline faunadb/client_test.go]: Could we change to something like `	s.queryAndDecode(f.EndsWith("faunadb", "db"), &b)` ?

File: faunadb/client_test.go [hunk 2]
```diff
@@ -1052,6 +1097,59 @@ func (s *ClientTestSuite) TestEvalTimeExpression() {
...
+
+func (s *ClientTestSuite) TestEvalTimeSubtractExpression() {
+	var t time.Time
+
+	s.queryAndDecode(f.TimeSubtract(f.Epoch(1, f.TimeUnitHour), 1, f.TimeUnitHour), &t)
+	s.Require().Equal(t, time.Unix(0, 0).UTC())
+
+	s.queryAndDecode(f.TimeSubtract(f.Epoch(16, f.TimeUnitSecond), 16, f.TimeUnitSecond), &t)
+	s.Require().Equal(t, time.Unix(0, 0).UTC())
```
Comment (evbruno) [inline faunadb/client_test.go]: Could we change those to something that isn't zero? (just being picky, I know...)

File: faunadb/client_test.go (hunk 1 above)
Comment (trevorsibanda) [reply faunadb/client_test.go]: I don't think this is necessarily a problem, an empty string is a valid string and an empty set is a subset of any set.

https://en.wikipedia.org/wiki/Empty_string

Comment (marrony) [reply faunadb/client_test.go]: I know it's not a problem, but you should at least test with a non-empty string to make sure you are covering "all" scenarios

File: faunadb/client_test.go [hunk 3]
```diff
@@ -941,6 +941,54 @@ func (s *ClientTestSuite) TestEvalCasefoldExpression() {
...
+	var b bool
+
+	s.queryAndDecode(f.ContainsStr("faunadb", "fauna"), &b)
+	s.Require().Equal(true, b)
+}
+
+func (s *ClientTestSuite) TestEvalContainsStrRegexExpression() {
+	var b bool
+
+	s.queryAndDecode(f.ContainsStrRegex("faunadb", `f(\w+)a`), &b)
```
Comment (marrony) [inline faunadb/client_test.go]: please, add the opposite case, when it doesn't contains the string

File: faunadb/client_test.go [hunk 4]
```diff
@@ -1802,6 +1903,73 @@ func (s *ClientTestSuite) TestEvalTruncExpression() {
...
+	counter := 0
+	maxCount := 27
+	data := make([]int, maxCount)
+
+	for counter < maxCount {
+		data[counter] = counter + 1
+		counter++
+	}
+
+	s.query(f.Foreach(data, f.Lambda("x", f.Create(f.Collection(aCollection), f.Obj{"data": f.Obj{"count": f.Var("x")}}))))
```
Comment (marrony) [inline faunadb/client_test.go]: you if are just `Count`ing the documents, you don't need to add `data`, just an empty object is okay

File: faunadb/client_test.go [hunk 5]
```diff
@@ -1802,6 +1903,73 @@ func (s *ClientTestSuite) TestEvalTruncExpression() {
...
+
+	s.queryAndDecode(f.Select(f.Arr{0}, f.Count(f.Paginate(f.Documents(f.Collection(aCollection))))), &i)
```
Comment (marrony) [inline faunadb/client_test.go]: please, add set version: `Count(Documents(Collection(...)))`

**Pull request merged**